lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -156,8 +387,9 @@ createSpecializedType(CanSILFunctionType SubstFTy, SILModule &M) const {
       SILType SILParamTy = SILType::getPrimitiveObjectType(PI.getType());
       // Indirect parameters are passed as owned, so we also need to pass the
       // direct parameter as owned (except it's a trivial type).
-      auto C = (SILParamTy.isTrivial(M) ? ParameterConvention::Direct_Unowned :
-                ParameterConvention::Direct_Owned);
+      auto C = ((!SILParamTy.hasTypeParameter() && SILParamTy.isTrivial(M))
```

Comment:
I'm not 100% sure about this. If it is trivial as a SILType, shouldn't it be possible load it and pass by-value?